File: 0846-hand-of-straights/0846-hand-of-straights.cpp
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int groupSize) {
        multiset<int>mst;
        for (auto it : hand)mst.insert(it);
        while(mst.size()){
            int x = *mst.begin();
            for(int i = x ; i < x+groupSize;i++){
                auto it = mst.find(i);
                if(it == mst.end())return false;
                mst.erase(it);
            }
        }
        return true;
    }
};

File: 0846-hand-of-straights/0846-hand-of-straights.py
class Solution:
    def isNStraightHand(self, hand: List[int], groupSize: int) -> bool:
        st = Counter(hand)
        while len(st) > 0:
            mn = min(st)
            for i in range(mn,mn+groupSize):
                if i in st:
                    st[i]-=1
                else:
                    return False
                if st[i] == 0:
                    del st[i]
        return True